### PR TITLE
Prevent overwrite of Slim embedded options

### LINKED ISSUE #1925: Slim::Embedded options being reset?
Using Middleman 4.1.8, I have this in my config.rb:

    require 'slim'
    require 'redcarpet'

    # ... 

    markdown_config = {
      :renderer            => CustomMarkdownRenderer,
      :fenced_code_blocks  => true,
      :autolink            => true
    }
    set :markdown_engine, :redcarpet
    set :markdown, markdown_config
    Slim::Embedded.set_options markdown: markdown_config

These custom options work fine with *.md files, but not with *.slim files.

When I inspect `Slim::Embedded.options` from a Ruby block in my Slim templates, I find that `Slim::Embedded.options[:markdown]` has been set to:

    :context=>#<#<Class:0x007f8728f42610>:0x007f872564d0f8 @app=#<Middleman::Application:0x70109242212760>, ...

Is Middleman reassigning `Slim::Embedded.options`? How can I prevent this so that Slim will use the Markdown options that I specify?

### PR DESCRIPTION
**Pull request by svicalifornia**
Fixes #1925 and perhaps #1152 also.

### CODE REVIEW

**Comment by svicalifornia** [comment]
Travis reports failure in middleman-core/features/asset_host.feature. My changes should not have affected that test, and that test is passing in my local environment running Ruby 2.3.0p0.

**Comment by tdreyno** [comment]
Guess we don't need `context_hack` then. Can you remove that?

The asset_host one is fine. It's trying to test something that is essentially random, everyone so often, it fails.

**Comment by svicalifornia** [comment]
OK, I removed `context_hack`.

File: middleman-core/lib/middleman-core/step_definitions/server_steps.rb [hunk 1]
```diff
@@ -88,14 +88,17 @@
 end
 
 Then /^I should see "([^\"]*)"$/ do |expected|
+  puts page.body
```
Comment (tdreyno) [inline middleman-core/lib/middleman-core/step_definitions/server_steps.rb]: This necessary or debugging?

Comment (svicalifornia) [inline middleman-core/lib/middleman-core/step_definitions/server_steps.rb]: Sorry, I'll remove that.

**Comment by tdreyno** [comment]
Thank you

**Pull request merged**